manim/scene/scene.py

```diff
@@ -373,6 +378,17 @@ def add_mobjects_among(self, values):
         self.add(*filter(lambda m: isinstance(m, Mobject), values))
         return self
 
+    def add_mobjects_from_animations(self, animations):
+
+        curr_mobjects = self.get_mobject_family_members()
+        for animation in animations:
+            # Anything animated that's not already in the
+            # scene gets added to the scene
+            mob = animation.mobject
+            if mob not in curr_mobjects:
```

Comment:
Question: what type is `curr_mobjects`? If it's a list (and it may be a large list?), then repeatedly checking for membership might take a while. Can we turn it into a set?

Reply:
`curr_mobjects` is a list. 
We _could_ modify it, but it's part of the old functioning of manim (so changing it can cause problems, idk)
and I don't think its the right PR to do it as it's (I think) a change that can cause trouble and I don't want additional problems with this PR :D